game/hud/src/widgets/Welcome/index.tsx

```diff
@@ -0,0 +1,130 @@
+      if (currentDate < savedDelayDate) visibilityDefault = false;
+    } catch (error) {
+      console.log(error);
+    }
+
+    //**TODO: set a decent default message. API is sometimes slow to return.
+    this.state = {
+      visible: visibilityDefault,
+      message: null
+    };
+  }
+
+  onMessage = (data: WelcomeData) => {
+    if (data.message == '') return;
+    const welcomeMessage: JSX.Element[] = [];
+
+    //**TODO: Remove the following test message workaround (API needs to return full HTML)
+    // data.message = '<style>' +
+    //   ' .welcome-box { height: 430px; width: 780px; background-image: url("http://i.imgur.com/XLsZQjI.jpg"); background-repeat: no-repeat; background-origin: content-box; }' +
+    //   ' .welcome-msg-bg { position: absolute; top: 0px; left: 10px; height: 406px; width: 780px; background: linear-gradient(rgba(60,50,50,0.5), rgba(60,50,50,0.7)); padding: 0.7em; color: white; }' +
+    //   ' .welcome-msg { position: absolute; top: 2em; left: 2em; height: 77%; width: 94%; background: rgba(255,250,205,0.5); padding: 1em; color: rgba(51,51,0,1); border-radius: 5px; }' +
+    //   '</style>' +
+    //   '<div class="welcome-box"><div class="welcome-msg-bg"><div class="welcome-msg">' +
+    //   '<p>Welcome to the Camelot Unchained Alpha server! Please read this message carefully, as the content is regularly updated.</p>' +
+    //   '<p><b>NOTE:</b> This is an "old-school" Alpha build of the Camelot Unchained client and server. This build is used for testing and demonstration purposes only. This should not be considered a reflection of the final game. Expect stability issues and drastic daily changes/improvements.</p>' +
+    //   '<p><u>Known Issues:</u><br /> - Animations are not final. They will be improved!<br /> - Movement speeds are not final.<br /> - Abilities are disabled.</p>' +
+    //   '<p>To view the Alpha Manual: <a href="https://forums.camelotunchained.com/topic/10538-alpha-manual-v042-updated-27-july-2015/" target="_new">CLICK HERE</a><br />' +
+    //   'To report bugs: <a href="https://forums.camelotunchained.com/forum/212-current-bugs-extended-alpha/" target="_new">CLICK HERE</a><br />' +
+    //   'To provide feedback/suggestions: <a href="https://forums.camelotunchained.com/forum/194-alpha-feedback-suggestions/" target="_new">CLICK HERE</a></p>' +
+    //   '<p>Try pressing <b style="color: rgba(110,45,23,1); font-size: 125%;">V</b> to create new abilities and <b style="color: rgba(110,45,23,1); font-size: 125%;">B</b> to open your ability book. Clicking the lock icon (lower-right corner) allows resizing and moving UI widgets. ' +
+    //   'For some helpful backer-supplied tips, type <b style="color: rgba(110,45,23,1); font-size: 125%;"><u>!tips</u></b> in the chat window.</p>' +
+    //   '</div></div></div>';
+
+    welcomeMessage.push(
+      <div key='100' dangerouslySetInnerHTML={{__html: data.message}} />
```

Comment:
Should we run the response data.message through something like https://github.com/cure53/DOMPurify before injecting html?  I don't imagine much harm can be done in this particular case via MITM or other response tampering, but do we have any hardcore web security experts on the case who can chime in and confirm?